### PR TITLE
[`CI update`] Try to use dockers and no cache

### PR DESCRIPTION
**Pull request by ArthurZucker**
# What does this PR do?

So we have: 
### 1. `setup_and_quality` 83% faster on average, instead of a total of 11min it's down to 2min36.
I think we can do even better with `check repo consistency` by reducing the size of the docker used! (purging nvidia-*, and else not needed for checks). Breakdown:
- 2x on check circle ci user (from 14sec -> 7s)
- 10x on check code quality (from 6min -> 30s)
- 2x on check repo consistency   (from 1min -> 30s)
- 2x on test fetching (from 1min -> 30s)

### 2. `run_tests` from 17min on average to 6min
image

before vs after:
image

### CODE REVIEW

**Comment by ArthurZucker** [comment]
I mostly need the approval of the security team at this point...

**Comment by ArthurZucker** [comment]
image

File: docker/custom-tokenizers.dockerfile
Comment (ArthurZucker) [inline docker/custom-tokenizers.dockerfile:21]: spacy not tested on the previous workflows either!

File: tests/models/layoutlmv2/test_image_processing_layoutlmv2.py
Comment (ArthurZucker) [inline tests/models/layoutlmv2/test_image_processing_layoutlmv2.py:115]:
```suggestion
        # the words and boxes were obtained with Tesseract 5.3.0
```

**Comment by ydshieh** [comment]
Wow, all ✅ ! I ❤️ seeing this. Will take a look this afternoon.
(But you can already remove `Draft` from the title 😉 )

File: docker/quality.dockerfile
Comment (LysandreJik) [inline docker/quality.dockerfile:4]:
```suggestion
RUN apt-get update && apt-get install -y time git 
```

File: setup.py
Comment (LysandreJik) [inline setup.py:355]: :)

File: .circleci/create_circleci_config.py
Comment (LysandreJik) [inline .circleci/create_circleci_config.py:64]:
```suggestion
```

File: .circleci/create_circleci_config.py
Comment (LysandreJik) [inline .circleci/create_circleci_config.py:44]:
```suggestion
```

File: .github/workflows/build-ci-docker-images.yml
Comment (LysandreJik) [inline .github/workflows/build-ci-docker-images.yml:14]:
```suggestion
    - cron: "6 0 * * *"
```

Comment (LysandreJik) [reply .github/workflows/build-ci-docker-images.yml:14]: so that it happens early in the morning for us

File: .github/workflows/build-ci-docker-images.yml
Comment (LysandreJik) [inline .github/workflows/build-ci-docker-images.yml:25]: As seen with you, TODO: handle on `main`

File: setup.py
Comment (LysandreJik) [inline setup.py:459]: todo

Comment (LysandreJik) [reply setup.py:459]: (in follow up pr)

File: src/transformers/models/tapas/modeling_tf_tapas.py
Comment (LysandreJik) [inline src/transformers/models/tapas/modeling_tf_tapas.py:64]:
```suggestion
        # On the first call, check whether a compatible version of TensorFlow is installed
```

**Review by LysandreJik — APPROVED** [review]

**Comment by LysandreJik** [comment]
(please wait for Yih-Dar's review before merging)

File: .circleci/parse_test_outputs.py
Comment (ydshieh) [inline .circleci/parse_test_outputs.py:56]: we don't need `default` here as `action` is specified.

File: .circleci/create_circleci_config.py
Comment (ydshieh) [inline .circleci/create_circleci_config.py:110]: in `MB` would be easier to read. So far it looks like

```bash
0.00573 GB libssl3
0.00611 GB dpkg
0.00640 GB linux-libc-dev
0.00683 GB bash
0.00728 GB perl-base
0.00745 GB libtsan2
0.00765 GB libasan8
```

File: .circleci/create_circleci_config.py
Comment (ydshieh) [inline .circleci/create_circleci_config.py:123]: maybe just a bit of explanation of `junit_family` (and others if you want) to people could understand easily without need to check the doc (if not really necessary)

File: .circleci/create_circleci_config.py
Comment (ydshieh) [inline .circleci/create_circleci_config.py:148]: is this to distribute the workload at the python file level ?

Comment (ydshieh) [reply .circleci/create_circleci_config.py:148]: FYI: I have seen that, once the list being way too long (and therefore pytest get a super long string in the arguments, it will fail

File: .circleci/create_circleci_config.py
Comment (ydshieh) [inline .circleci/create_circleci_config.py:190]: hmm, I don't remember why I added `|| true`, but I believe it's not necessary (at least not anymore)

File: .circleci/create_circleci_config.py
Comment (ydshieh) [inline .circleci/create_circleci_config.py:329]: are these 2 still necessary?

File: src/transformers/models/groupvit/modeling_tf_groupvit.py
Comment (ydshieh) [inline src/transformers/models/groupvit/modeling_tf_groupvit.py:74]: this look odd no?

File: src/transformers/models/tapas/modeling_tf_tapas.py
Comment (ydshieh) [inline src/transformers/models/tapas/modeling_tf_tapas.py:833]: Also I am not sure this is the best action to take.

File: src/transformers/models/groupvit/modeling_tf_groupvit.py
Comment (ydshieh) [reply src/transformers/models/groupvit/modeling_tf_groupvit.py:74]: If this is about 

> tfp is not defined even if installed. FIXME @Arthur in a followup PR!"

similar to the test, OK for me for now, but maybe also add the same comment here.

**Review by ydshieh — APPROVED** [review]
Thank you @ArthurZucker . This is really great work 🚀 .

Mostly some nit comments, and let discuss/think about the image push process after this being merged. 

(Please don't merge yet before I DM you on slack)

File: docker/consistency.dockerfile
Comment (ArthurZucker) [reply docker/consistency.dockerfile:1]: IMo this makes more sense, nightly can use 3.8 !

File: .circleci/create_circleci_config.py
Comment (ArthurZucker) [reply .circleci/create_circleci_config.py:214]: uv is just a lot faster than pip to resolve anything

File: docker/torch-light.dockerfile
Comment (ArthurZucker) [reply docker/torch-light.dockerfile:7]: We almost did the same before. I will updtae this one I add extra deps. The idea is that I HAVE to force CPU

File: .circleci/create_circleci_config.py
Comment (ArthurZucker) [reply .circleci/create_circleci_config.py:110]: OK will update!

File: .circleci/create_circleci_config.py
Comment (ArthurZucker) [reply .circleci/create_circleci_config.py:123]: Sure! junit is to prepare for the split that will come next. Junit 2 is incompatible

File: .circleci/create_circleci_config.py
Comment (ydshieh) [reply .circleci/create_circleci_config.py:214]: no, i mean why need `venv`. I understand the second part `uv pip install .`

File: .circleci/create_circleci_config.py
Comment (ArthurZucker) [reply .circleci/create_circleci_config.py:148]: Ahhh that a good catch. I might revert.
Same here the splitting can be better if it's on sub files not folders

Comment (ArthurZucker) [reply .circleci/create_circleci_config.py:148]: and circle ci suppo9rt that better

Comment (ydshieh) [reply .circleci/create_circleci_config.py:148]: OK, as long as it works so far, no need to revert. let's see

File: src/transformers/models/groupvit/modeling_tf_groupvit.py
Comment (ArthurZucker) [reply src/transformers/models/groupvit/modeling_tf_groupvit.py:74]: Yes even if tfp is installed it's not detecting it

File: src/transformers/models/tapas/modeling_tf_tapas.py
Comment (ArthurZucker) [reply src/transformers/models/tapas/modeling_tf_tapas.py:833]: same as above the is available is not always write

File: .circleci/create_circleci_config.py
Comment (ArthurZucker) [reply .circleci/create_circleci_config.py:214]: You cannot use uv otherwise!

File: docker/torch-light.dockerfile
Comment (ArthurZucker) [reply docker/torch-light.dockerfile:9]: That's a good catch, adding torch-speech now

File: .github/workflows/build-ci-docker-images.yml
Comment (ArthurZucker) [reply .github/workflows/build-ci-docker-images.yml:25]: That is here to check the status of the previous workflow run. We see if it 's needed

File: .circleci/create_circleci_config.py
Comment (ArthurZucker) [reply .circleci/create_circleci_config.py:188]: Yes! You did not want to make it fail, but it's actually better to let it fail, but use `when: always` on the condition of the next 3 steps that check the output

File: .circleci/create_circleci_config.py
Comment (ArthurZucker) [reply .circleci/create_circleci_config.py:327]: Mmm yes I did not create the docker for this one, will do in a follow up PR if needed.

**Pull request merged**
